### PR TITLE
WooCommerce: Add the order list view

### LINKED ISSUE #13654: WooCommerce: Orders list view
Create the orders list view for WooCommerce using data from the WooCommerce API.

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #13654 

screen shot 2017-06-20 at 9 47 49 am

This adds the basic orders list view. We've decided pagination will be part of v2, so right now you can only see a max of 100 orders. This particular PR just gets the basic display in, I'll add follow-ups for better currency display, and for the split status bar (in https://github.com/Automattic/wp-calypso/issues/13654#issuecomment-306613311).

To test:

- Visit `https://calypso.live/store/orders/$site`
- You should see the list of orders on your site

### CODE REVIEW

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+			</ul>
+		);
+	}
+
+	render() {
+		const { orders, ordersLoading, ordersLoaded, translate } = this.props;
+		const headers = (
+			<TableRow>
+				<TableItem isHeader>
+					<FormInputCheckbox aria-label={ translate( 'Select All', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/orders/orders-list.js]: Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Select all' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string already exists without a **context**. Only add a context if the meaning of the string is very specific.

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+				return <span className={ classes }>{ translate( 'Failed' ) }</span>;
+		}
+	}
+
+	renderOrderItems = ( order, i ) => {
+		const { translate, moment, siteId } = this.props;
+		return (
+			<TableRow key={ i }>
+				<TableItem className="orders__table-checkbox">
+					<FormInputCheckbox aria-label={ translate( 'Select order %(order)s', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/orders/orders-list.js]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+			case 'processing':
+				return <span className={ classes }>{ translate( 'Processing' ) }</span>;
+			case 'on-hold':
+				return <span className={ classes }>{ translate( 'On hold' ) }</span>;
+			case 'completed':
+				return <span className={ classes }>{ translate( 'Completed' ) }</span>;
+			case 'cancelled':
+				return <span className={ classes }>{ translate( 'Cancelled' ) }</span>;
+			case 'refunded':
+				return <span className={ classes }>{ translate( 'Refunded' ) }</span>;
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/orders/orders-list.js]: Hi! I've found a possible matching string that has already been translated **37** times:
`translate( 'Refunded To' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+		const classes = `orders__item-status is-${ status }`;
+		switch ( status ) {
+			case 'pending':
+				return <span className={ classes }>{ translate( 'Pending payment' ) }</span>;
+			case 'processing':
+				return <span className={ classes }>{ translate( 'Processing' ) }</span>;
+			case 'on-hold':
+				return <span className={ classes }>{ translate( 'On hold' ) }</span>;
+			case 'completed':
+				return <span className={ classes }>{ translate( 'Completed' ) }</span>;
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/orders/orders-list.js]: Hi! I've found a possible matching string that has already been translated **52** times:
`translate( '&laquo; completed' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
Awesome!!! Going to style this up, but one comment before I do: 

Since we aren't including any bulk editing for v1, let's get rid of the column of checkboxes.

**Comment by kellychoffman** [comment]
When there are no orders, can we use `EmptyContent` [#](https://wpcalypso.wordpress.com/devdocs/design/empty-content) like we do in the product list view? (Apologize for not putting this in the design)

Copy:

```Orders will appear here as they come in.

[Manually add an order]```

**Comment by ryelle** [comment]
I've added the EmptyContent component, and also split out the the status labels — I'm not sure I got the labels right, so I could use a second look (from anyone) at how they're generated (https://github.com/Automattic/wp-calypso/pull/14871/commits/c00ba35d3fdbbe9ea4d845d7e98ca2e7b90b91e4).

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+	}
+
+	getOrderStatus = ( status ) => {
+		const { translate } = this.props;
+		const classes = `orders__item-status is-${ status }`;
+		let paymentLabel;
+		let shippingLabel;
+		switch ( status ) {
+			case 'pending':
+				shippingLabel = translate( 'New order' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/orders/orders-list.js]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Save New Order' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-order-list)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
Split labels are looking good! 

Could you remove the sectionNav above emptyContent?

**Comment by allendav** [comment]
I was surprised to see two states for this order on octopus - was expecting just "on-hold" for it

screen shot 2017-06-20 at 3 58 12 pm

**Comment by allendav** [comment]
@kellychoffman / @jameskoster - would the order date be a useful place for moment instead? (e.g. 1 hour ago, 1 day ago, etc etc)

screen shot 2017-06-20 at 4 02 35 pm

File: client/extensions/woocommerce/app/orders/index.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
```
Comment (allendav) [inline client/extensions/woocommerce/app/orders/index.js]: Would the fallback selector be more robust here?  https://github.com/Automattic/wp-calypso/blob/e1f9cddd82b2ca92e1e3f20f7cbec36812ea088b/client/extensions/woocommerce/state/sites/selectors.js#L16

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+				paymentLabel = translate( 'Payment pending' );
+				break;
+			case 'processing':
+				shippingLabel = translate( 'New order' );
+				paymentLabel = translate( 'Paid in full' );
+				break;
+			case 'on-hold':
```
Comment (allendav) [inline client/extensions/woocommerce/app/orders/orders-list.js]: Ah, now I see why we were getting two labels for some orders. Nevermind my other comment :P

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+				</Table>
+				{ this.renderPagination() }
+			</div>
+		);
+	}
+}
+
+export default connect(
+	state => {
+		const siteId = getSelectedSiteId( state );
```
Comment (allendav) [inline client/extensions/woocommerce/app/orders/orders-list.js]: See fallback suggestion above

**Comment by allendav** [comment]
The order state needs a little love on narrowish screens:

screen shot 2017-06-20 at 4 15 12 pm

cc @kellychoffman @jameskoster

**Comment by allendav** [comment]
If I:

- go to http://calypso.localhost:3000/store/orders/{siteslug}
- indexedDB.deleteDatabase( 'calypso' );
- localStorage.clear()
- refresh (Command-Shift-R)

The body fails to render after the sidebar eventually does and I get:

screen shot 2017-06-20 at 4 21 35 pm

**Comment by kellychoffman** [comment]
> The order state needs a little love on narrowish screens:

I tried a bunch of things, but can't figure out how to have a max-width on the table so that it horizontally scrolls on screens smaller than 960px. We have this on the variation table in add-product, but I can't get it to apply here. I think its because the variations table is within a `Card`?

**Comment by jameskoster** [comment]
@ryelle could we make it so that the entire row links to the order rather than just the number?

**Comment by ryelle** [comment]
> @ryelle could we make it so that the entire row links to the order rather than just the number?

Probably, I can look into that when I do the single-view PR

**Comment by ryelle** [comment]
I've switch to the fallback selector and fixed the undefined site issues, but I think the other concerns can be handled in another PR — would love to merge this before I need to rebase it again 😄

File: client/extensions/woocommerce/app/orders/orders-list.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+						{ `${ order.billing.first_name } ${ order.billing.last_name }` }
+					</span>
+				</TableItem>
+				<TableItem className="orders__table-date">
+					{ moment( order.date_modified ).format( 'LLL' ) }
+				</TableItem>
+				<TableItem className="orders__table-status">
+					{ this.getOrderStatus( order.status ) }
+				</TableItem>
+				<TableItem className="orders__table-total">
```
Comment (kellychoffman) [inline client/extensions/woocommerce/app/orders/orders-list.js]: I faked this for now, but would be good to get in another PR.

**Comment by kellychoffman** [comment]
> @ryelle could we make it so that the entire row links to the order rather than just the number?

This is also done in Products list. These tables should be as consistent as possible, but fine to do in another PR.

**Comment by kellychoffman** [comment]
> I tried a bunch of things, but can't figure out how to have a max-width on the table so that it horizontally scrolls on screens smaller than 960px. We have this on the variation table in add-product, but I can't get it to apply here. I think its because the variations table is within a Card?

Followup on this: Jay addressed this in https://github.com/Automattic/wp-calypso/pull/15351

**Comment by ryelle** [comment]
>This is also done in Products list. These tables should be as consistent as possible, but fine to do in another PR.

I just want to double check the accessibility of linking an entire table row, and possibly I'll update the shared table component to make this super-easy in all our tables. But that's why I said another PR 🙂
